read-fonts/src/tables/ift.rs

```diff
@@ -0,0 +1,344 @@
+            max_entry_index: self.max_entry_index(),
+        }
+    }
+}
+
+struct GidToEntryIter<'a> {
+    glyph_map: Option<GlyphMap<'a>>,
+    glyph_count: u32,
+    gid: u32,
+}
+
+impl<'a> Iterator for GidToEntryIter<'a> {
+    type Item = (GlyphId, u16);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let glyph_map = self.glyph_map.as_ref()?;
+
+        let cur_gid = self.gid;
+        self.gid += 1;
+
+        if cur_gid >= self.glyph_count {
+            return None;
+        }
+
+        let index = cur_gid as usize - glyph_map.first_mapped_glyph() as usize;
+        glyph_map
+            .entry_index()
+            .get(index)
+            .ok()
+            .map(|entry_index| (cur_gid.into(), entry_index.0))
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct FeatureEntryMapping {
+    pub matched_entries: RangeInclusive<u16>,
+    pub new_entry_index: u16,
+    pub feature_tag: Tag,
+}
+
+struct EntryMapIter<'a, T>
+where
+    T: Iterator<Item = Result<FeatureRecord, ReadError>>,
+{
+    data: &'a [u8],
+    feature_record_it: Option<T>,
+    current_feature_record: Option<FeatureRecord>,
+    remaining: u16,
+    max_entry_index: u16,
+}
+
+impl<'a, T> EntryMapIter<'a, T>
+where
+    T: Iterator<Item = Result<FeatureRecord, ReadError>>,
+{
+    fn empty(data: &'a [u8]) -> Self {
+        EntryMapIter {
+            data,
+            feature_record_it: None,
+            current_feature_record: None,
+            remaining: 0,
+            max_entry_index: 0,
+        }
+    }
+}
+
+impl<T> Iterator for EntryMapIter<'_, T>
+where
+    T: Iterator<Item = Result<FeatureRecord, ReadError>>,
+{
+    type Item = FeatureEntryMapping;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // TODO(garretrieger): current spec text for this is wrong and doesn't match what were doing here. Update
+        //                     the spec to match this implementation.
+        let feature_record_it = self.feature_record_it.as_mut()?;
+
+        if self.current_feature_record.is_none() || self.remaining == 0 {
+            let Some(Ok(feature_record)) = feature_record_it.next() else {
+                self.feature_record_it = None;
+                return None;
+            };
+
+            self.remaining = feature_record.entry_map_count().get();
+            self.current_feature_record = Some(feature_record);
+        }
+
+        let data = FontData::new(self.data);
+        let (Some(feature_record), Ok(entry_record)) = (
+            self.current_feature_record.clone(),
+            EntryMapRecord::read(data, self.max_entry_index),
+        ) else {
+            self.feature_record_it = None;
+            return None;
+        };
+
+        let new_entry_index = feature_record.first_new_entry_index.get()
+            + (feature_record.entry_map_count().get() - self.remaining);
+        self.remaining -= 1;
+
+        let size = U8Or16::compute_size(&self.max_entry_index).ok()? * 2;
+        self.data = &self.data[size..];
```

Comment:
Is this guaranteed to be in bounds?

Changing to `self.data.get(size..)?` would prevent a panic just in case.